skimage/measure/fit.py

```diff
@@ -668,7 +668,8 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
         return np.inf
     inlier_ratio = n_inliers / n_samples
     nom = max(_EPSILON, 1 - probability)
-    denom = max(_EPSILON, 1 - inlier_ratio ** min_samples - _EPSILON)
+    denom = max(_EPSILON, 1 - inlier_ratio ** min_samples)
+    denom = min(denom, 1 - _EPSILON)
```

Comment:
```suggestion
    # Avoid log(1) below turning into -inf
    denom = min(denom, 1 - _EPSILON)
```

Reply:
I think `denom = np.clip(1 - inlier_ratio**min_samples, _EPSILON, 1 - _EPSILON)`  is shorter an more expressive.

Reply:
Agreed. Though I think the 2nd and 3rd argument need to be switched. I took the liberty to add your suggestion in https://github.com/scikit-image/scikit-image/pull/7065/commits/f450f4ce4ccd2ba93520a8d08bd62be025bc68ec. I added you as a co-author but didn't know an email associated with your account so GitHub will not pick that up. Let me know if I should modify the message.

Reply:
> Though I think the 2nd and 3rd argument need to be switched.

I do not understand what you mean. https://github.com/scikit-image/scikit-image/pull/7065#discussion_r1710976764 is the same as implemented in https://github.com/scikit-image/scikit-image/commit/f450f4ce4ccd2ba93520a8d08bd62be025bc68ec, right?

Additionally, I would also limit the nominator as in https://github.com/scikit-image/scikit-image/pull/7490/files. That way it is symmetric and (de-)nominator are both in (0, 1).

I added an email address to my GitHub account.

Reply:
Sorry, I was misreading your suggestion. I agree that guarding the nominator in a similar manner would be useful. Would you be willing to make a small PR for that? (Feel free to ping me!) That way you'll also get commit credits. :slightly_smiling_face:

Reply:
Good suggestion. I opened a PR: https://github.com/scikit-image/scikit-image/pull/7496